probe/endpoint/reporter.go

```diff
@@ -33,10 +33,11 @@ var SpyDuration = prometheus.NewSummaryVec(
 
 // NewReporter creates a new Reporter that invokes procspy.Connections to
 // generate a report.Report that contains every discovered (spied) connection
-// on the host machine, at the granularity of host and port. It optionally
-// enriches that topology with process (PID) information.
-func NewReporter(hostID, hostName string, includeProcesses bool) tag.Reporter {
-	return &reporter{
+// on the host machine, at the granularity of host and port. That information
+// is stored in the Endpoint topology. It optionally enriches that topology
+// with process (PID) information.
+func NewReporter(hostID, hostName string, includeProcesses bool) *Reporter {
```

Comment:
We could define Reporter and Tagger in another package, but it seems needless, since every package {host,endpoint,process,...} would have 1 Reporter and Tagger struct.